components/authz-service/storage/postgres/migration/sql/52_query_returns_condition_type.up.sql

```diff
@@ -0,0 +1,194 @@
+BEGIN;
+
+CREATE OR REPLACE FUNCTION
+  query_staged_and_applied_rules(_project_filter TEXT[])
+  RETURNS setof json AS $$
+
+  -- fetch staged rules
+  SELECT json_build_object(
+    'id', r.id,
+    'project_id', r.project_id,
+    'name', r.name,
+    'type', r.type,
+    'deleted', r.deleted,
+    'status', 'staged',
+    'conditions', json_agg(
+      json_build_object(
+        'value', rc.value,
+        'operator', rc.operator,
+        'attribute', rc.attribute,
+        -- condition type always matches the rule type
+        'type', r.type
+      )
+    )
+  ) AS rules
+  FROM iam_staged_project_rules AS r
+  INNER JOIN iam_staged_rule_conditions AS rc ON rc.rule_db_id=r.db_id
+  WHERE projects_match_for_rule(project_id, _project_filter)
+  GROUP BY r.id, r.project_id, r.name, r.type, r.deleted
+
+  UNION ALL
+
+  -- fetch applied rules
+  SELECT json_build_object(
+    'id', r.id,
+    'project_id', r.project_id,
+    'name', r.name,
+    'type', r.type,
+    'deleted', false,
+    'status', 'applied',
+    'conditions', json_agg(
+      json_build_object(
+        'value', rc.value,
+        'operator', rc.operator,
+        'attribute', rc.attribute,
+        -- condition type always matches the rule type
+        'type', r.type
+      )
+    )
+  ) AS rules
+  FROM iam_project_rules AS r
+  INNER JOIN iam_rule_conditions AS rc ON rc.rule_db_id=r.db_id
+  WHERE projects_match_for_rule(project_id, _project_filter)
+  GROUP BY r.id, r.project_id, r.name, r.type;
+
+$$ LANGUAGE sql;
+
+CREATE OR REPLACE FUNCTION
+  query_staged_or_applied_rule(_rule_id TEXT, _project_filter TEXT[])
+  RETURNS json AS $$
+
+  -- check for applied rule
```

Comment:
```suggestion
  -- check for staged rule
```
(You did not introduce this bug, but you've got the power to fix it anyway. 😁  )

Reply:
decided to scrap this migration after all :o